safehttp/response_writer.go

```diff
@@ -25,18 +25,22 @@ type ResponseWriter struct {
 
 	// Having this field unexported is essential for security. Otherwise one can
 	// easily overwrite the struct bypassing all our safety guarantees.
-	header  Header
-	written bool
+	header           Header
+	appliedInterceps []AppliedInterceptor
+	req              *IncomingRequest
+	written          bool
 }
 
 // NewResponseWriter creates a ResponseWriter from a safehttp.Dispatcher, an
 // http.ResponseWriter and a list of interceptors associated with a ServeMux.
-func NewResponseWriter(d Dispatcher, rw http.ResponseWriter) *ResponseWriter {
+func NewResponseWriter(d Dispatcher, rw http.ResponseWriter, req *IncomingRequest, interceps []AppliedInterceptor) *ResponseWriter {
```

Comment:
It doesn't seem right that the `ResponseWriter` has direct control over the interceptors. It should ask the mux or the handler wrapper that has the interceptors to do the work.